slides/week5/week5.org

```diff
@@ -0,0 +1,91 @@
+#+TITLE: Week 5
+#+AUTHOR: Kristaps Berzinch, Raymond Ortiz
+#+EMAIL: kristaps@robojackets.org, rortiz9@gatech.edu
+
+* What are we doing today?
+- Classes in C++
+- Handling merge conflicts in =git=
+
+* What are classes?
+- Definitions for objects, i.e. what data and methods they have
+ - In other words, a blueprint for how to create an object
+- An object is called an *instance* of a class
+ - An object has all the functions and data structures defined in its class
+ - Data schema is the same across all instances of a class, but values differ
+ - It's interchangeable with other objects from the same class
+
+* A simple example
+#+BEGIN_NOTES
+A class can have some data and some methods, which can either be public or private. Private means only visible to the class itself, public means visible to any other file. Drop to a terminal and use/modify this class.
+#+END_NOTES
+#+BEGIN_SRC c++
+class example {
+    private:         // private is also implied if you don't specify visibility
+        int a;
+    public:
+        int getA() { return a; }
+};
+#+END_SRC
+
+* Why bother?
+- Software projects are easier to understand when related data and functions are grouped together
+- Abstracting data behind getters/setters allows you to validate inputs from other parts of your application
+- Abstracting complex tasks into class methods enables simpler, easier to read high-level code
+- Classes can be extended to add functionality with minimal extra code using inheritance
+
+* Static members
+#+BEGIN_NOTES
+ Drop to a terminal and use this class.
+#+END_NOTES
+- For when you want data or functions to be part of your class, but they don't need to be "attached" to an instance of the class
```

Comment:
This wording seems a little more confusing than explicitly saying that all instances of a class read from and write to the same static variable